In [20]:
# Importing Liberaries
import pandas as pd
import scipy.stats as st
import math

# 1. First load the data in pandas

In [2]:
df = pd.read_csv('catalyst.csv')
df

Catalyst1  Catalyst2
0      91.50      89.19
1      94.18      90.95
2      92.18      90.46
3      95.39      93.21
4      91.79      97.19
5      89.07      97.04
6      94.72      91.07
7      89.21      92.75

# 2. Calculate the sample mean and sample standard deviation for both the catalyst

In [3]:
# For column-Catalyst 1 
mu1 = df['Catalyst1'].mean()
std1 = df['Catalyst1'].std()
print(f' Mean for Catalyst1 = {mu1}\n Standard Deviation for Catalyst1 = {std1}')

 Mean for Catalyst1 = 92.25500000000001
 Standard Deviation for Catalyst1 = 2.385018718105646


In [4]:
# For column-Catalyst 2 
mu2 = df['Catalyst2'].mean()
std2 = df['Catalyst2'].std()
print(f' Mean for Catalyst2 = {mu2}\n Standard Deviation for Catalyst2 = {std2}')

 Mean for Catalyst2 = 92.73249999999999
 Standard Deviation for Catalyst2 = 2.983453176629871


# 3. The number of rows for each catalyst

In [5]:
# for Catalyst1
row_count1 = df['Catalyst1'].count()
print(f"No. of Rows for Catalyst1 = {row_count1}")

No. of Rows for Catalyst1 = 8


In [6]:
# for Catalyst2
row_count2 = df['Catalyst2'].count()
print(f"No. of Rows for Catalyst2 = {row_count2}")

No. of Rows for Catalyst2 = 8


# 4. Using 8 Step Hypothesis Process

In [10]:
""" 1st Step The Parameter of Interest are mu1 and mu2, the mean process yield using catalyst 1 and 2 respectively. We want to know if mu1-mu2 = 0 """

mu = mu1-mu2
if(mu == 0):
    print(" mu1 - mu2 = 0 Therefore it satisfy 1st Condition")
else:
    print("1st step is not satisfied!!!!")

1st step is not satisfied!!!!


In [11]:
""" 2nd Step Null Hypothesis Ho: mu1 - mu2 = 0 OR mu1 = mu2 """

if(mu1 == mu2):
    print("mu1 - mu2 = 0 AND mu1 = mu2 Therefore the null hypothesis condition is satisfied")
else:
    print("Null Hypothesis Contition is not satisfied")

Null Hypothesis Contition is not satisfied


In [12]:
""" 3rd Step Alternate Hypothesis Ha: mu1 != mu2 """
if (mu1 != mu2):
    print("Alternative Hypothesis Condition is Satisfied")
else:
    print("Alternative Hypothesis Condition is not Satisfied")

Alternative Hypothesis Condition is Satisfied


In [13]:
""" 4th Step alpha = 0.05 """
print("Alpha = 0.05")

Alpha = 0.05


*** """ 5th Step the Test Statistics """ ***


t0 = (mu1 - mu2 - 0)/Sp*((1/n1)+(1/n2)

In [15]:
""" 6th Step Reject Ho: If t0 > t0.025,14 = 2.145 OR If t0 < -t0.025,14 = -2.145 """ 
st.t.ppf(1.95/2,14)

2.1447866879169273

In [42]:
""" 7th Step Calculation """
Test_Stat = st.ttest_ind(df['Catalyst1'], df['Catalyst2'])
print(f"Test Stats Comes out to be  =  {Test_Stat}")

Test Stats Comes out to be  =  Ttest_indResult(statistic=-0.3535908643461798, pvalue=0.7289136186068217)


***""" 8th Step CONCLUSION """***

Conclusion: Since -2.147 < t0 = -0.35 < 2.145  the null hypothesis cannot be rejected. That is at the 0.05 significance level, we do not have strong evidence to conclude that catalyst 2 results in a mean yield that differs from the mean yield when catalyst 1 is used.


# Summary 

- First we added important Liberaries
- Secondly we loaded the data and read the data.
- Then we calculated the Mean And Standard Deviation for both catalyst which comes out to be 
    
    Catalyst 1
    
        Mean for Catalyst1 = 92.25500000000001
    
        Standard Deviation for Catalyst1 = 2.385018718105646
    
    Catalyst 2 
    
        Mean for Catalyst2 = 92.73249999999999
    
        Standard Deviation for Catalyst2 = 2.983453176629871

- Then we found out the no. of rows for both columns which comes out tu be 8 for both columns.
- At last we Used out Eight Step Hypothesis Step in which we got to know the test stats which comes out to be 

    Test Stats Comes out to be  =  Ttest_indResult(statistic=-0.3535908643461798, pvalue=0.7289136186068217)